benchmark/single-source/KeyPathPerformanceTests.swift

```diff
@@ -0,0 +1,1009 @@
+//===--- KeyPathPerformanceTests.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import TestsUtils
+
+public let benchmarks = [
+  BenchmarkInfo(
+    name: "KeyPathNestedStructs",
+    runFunction: run_KeyPathNestedStructs,
+    tags: [.validation, .api],
+    setUpFunction: setupKeyPathNestedStructs
+  ),
+  BenchmarkInfo(
+    name: "KeyPathDirectAccess",
+    runFunction: run_testDirectAccess,
+    tags: [.validation, .api],
+    setUpFunction: setupSingleton
+  ),
+  BenchmarkInfo(
+    name: "KeyPathReadPerformance",
+    runFunction: run_testKeyPathReadPerformance,
+    tags: [.validation, .api],
+    setUpFunction: setupSingleton
+  ),
+  BenchmarkInfo(
+    name: "KeyPathWritePerformance",
+    runFunction: run_testKeyPathWritePerformance,
+    tags: [.validation, .api],
+    setUpFunction: setupSingleton
+  ),
+]
+
+// Number of iterations for each benchmark.
+let numberOfElementsForNestedStructs = 20000
+
+// Make it easy to switch between primitives of different types (Int, Float, Double, etc.).
+typealias ElementType = Double
+
+// Variables used across the benchmarks
+let expectedIntForNestedStructs = 3
+let arrayHolder = FixedSizeArrayHolder()
```

Comment:
`arrayHolder` is not used anywhere. What's the purpose of this?
`FixedSizeArrayHolder.shared` should be the right thing to use.

Reply:
This was part of an earlier design and has been removed in the latest commit.